### DISCUSSION TITLE
use postprocessor value in ParsedMaterial?

### DISCUSSION THREAD

**Discussion by jessecarterMOOSE** [idaholab/moose#15784] [Q&A General] [answered]
I'd like to use a postprocessor value in my ParsedMaterial. I tried to put the name in the 'args' list but that didn't work because it seems to be looking for variables. I suppose I could make an AuxVariable that holds the postprocessor value everywhere, but that seems to be a bit overkill. Is there a better way?

**Comment by dschwen** [comment]
No, there isn't a better way yet, sorry. It would not be hard to add that capability though.

**Comment by tophmatthews** [comment]
What about holding the postprocessor as a function instead of an aux? Still an extra step, but at least it's not an aux defined everywhere?

**Reply by jessecarterMOOSE** [reply]
Makes sense 👍

**Reply by dschwen** [reply]
Well, that's a possibility, but it doesn't really answer the question, because to get a Function value into a `ParsedMaterial` you'd have to use another step - either `FunctionAux` or `GenericFunctionMaterial` :-D (the latter would be more memory efficient)

**Reply by tophmatthews** [reply]
oh ya...what a pain...

**Comment by dschwen** [comment, marked as answer]
To make this a _markable_ answer: Use a `GenericFunctionMaterial` to capture the PostProcessor value and add it to `ParsedMaterial` using the `material_property_names` parameter.

**Reply by tophmatthews** [reply]
Or bug @dschwen enough until he adds the capability you need :)

**Reply by jessecarterMOOSE** [reply]
I ended up using @tophmatthews 's suggestion because what I was actually, _secretly_ doing was using the material to mark a region either 0 or 1 and then integrating that material with a postprocessor to find the area fraction of that marker, then normalize that value by another postprocessor. Dividing the area fraction postprocessor value by the other in a `ParsedFunction` was all that was required. Guess I should have been more clear on what I was after. Though to @dschwen 's credit, using the `GenericFunctionMaterial` would have worked too. Both require an extra step.

What would also be neat is a `ParsedPostprocessor` where I can just divide the two postprocessors directly instead of making the intermediate function. Then you might be able to get rid of `Difference`, `LinearCombination`, and `Scale` postprocessors, and maybe a few others.
